### PR TITLE
Using url param

### PR DESCRIPTION
**Pull request by jhaniv**
update code to use the url param
updated the tutorial accordingly

### CODE REVIEW

File: src/provider/handler/getPosts.js
Comment (francois-codes) [inline src/provider/handler/getPosts.js]: please return reject here, for consistency

File: src/provider/handler/getPosts.js
Comment (francois-codes) [inline src/provider/handler/getPosts.js]: you don't really need the promise constructor here. You can run the code (if you want to reject you can do `return Promise.reject({...});`

just parse your url here, and when it's formed do `return axios.get('...').then()...`

File: src/provider/handler/getPosts.js
Comment (francois-codes) [inline src/provider/handler/getPosts.js]: please add [prettier](https://github.com/prettier/prettier) to the project for consistent code formatting, and enable auto format on save in your editor

Comment (francois-codes) [inline src/provider/handler/getPosts.js]: instead of allocating an array for this, you can also do something like
```
return Promise.all(result.entry.map(result => axios.get(...))
  .then(results => ...)
```

what you can extract is the mapResultToPromise function, for readability, and then you can can simply invoke `Promise.all(result.entry.map(mapResultToPromise))`

File: src/provider/handler/getPosts.js
Comment (francois-codes) [inline src/provider/handler/getPosts.js]: you're going to do this on all your axios calls. 
So I invite you to add a helpers or utils file that includes a function that wraps this, like : 
```
const fetchUrl = (url, options) => axios.get(url, options)
  .then(response => {
     if (!response.data && ...) {
        throw new Error('....'); // will go to .catch block, so you can customize the error message you return
     }
     return response.data 
  });
```

then you just use
```
fetch(url)
  .then(handleData)
  .catch(error => Promise.reject({ message: '...', code: 500 }))
```

try to use as many small blocks as you can so you can easily compose functions and chain promises. the result is more readable

File: src/provider/handler/mappers/postMapper.js
Comment (francois-codes) [inline src/provider/handler/mappers/postMapper.js]: same here, you may want to reuse this logic somewhere else - you could extract a `getPostFullTitle` function or something

File: src/provider/handler/mappers/categoryMapper.js
Comment (francois-codes) [inline src/provider/handler/mappers/categoryMapper.js]: please use deconstruction and shorthand property assignment as much as possible
```
const { id, name: title } = category
return { id, title };
```

**Review by francois-codes — CHANGES_REQUESTED** [review]
@jhaniv I made rather detailed comments so you can adapt to the coding style we try to enforce at applicaster. Overall it will gain in visibility

**Comment by jhaniv** [comment]
made a bigger refactoring, especially around the whole media mapping part.
hope I didn't miss anything.
Also, I'm not sure that my prettier settings matches your requirement, so let me know.

**Review by lomudi — APPROVED** [review]
LGTM 💯 
Please also wait for @f-roland approval. thx.

File: src/provider/handler/mappers/mapPostMediaRequest.js
Comment (francois-codes) [inline src/provider/handler/mappers/mapPostMediaRequest.js]: why are you returning a negative id here ?
if you want to exclude this result without crashing the `Promise.all` I believe you can just return null

File: src/provider/handler/mappers/postMapper.js
Comment (francois-codes) [inline src/provider/handler/mappers/postMapper.js]: every time you are using the `forEach` / `array.push()` combo, you can usually either use `map` or `reduce` function

I would do something like : 
```
const mediaGroups = mediaItems
   .filter(removeItemsYouDontWant)
   .map(transformMediaItemObject)

```

File: src/provider/handler/mappers/postMapper.js
Comment (francois-codes) [inline src/provider/handler/mappers/postMapper.js]: is this working ? I'm not sure we decode URIs when calling children links

**Review by francois-codes — CHANGES_REQUESTED** [review]
some of the previous comments were not addressed (the one with deconstructing)
I also added a couple more

File: src/provider/handler/mappers/mapPostMediaRequest.js
Comment (jhaniv) [reply src/provider/handler/mappers/mapPostMediaRequest.js]: I can't return null but I can return {id: null}

File: src/provider/handler/mappers/postMapper.js
Comment (jhaniv) [reply src/provider/handler/mappers/postMapper.js]: I thought that's what the documentation said about article.html - so I should just return the html as it is?

**Comment by jhaniv** [comment]
@f-roland where did I miss the destructuring?

File: src/provider/handler/mappers/postMapper.js
Comment (francois-codes) [reply src/provider/handler/mappers/postMapper.js]: Ok, now I understand.

HTML content may be escaped somehow - we will need to check with native developers, but I doubt it is done with the function that escapes special characters from url parameters

File: src/provider/handler/mappers/categoryMapper.js
Comment (francois-codes) [reply src/provider/handler/mappers/categoryMapper.js]: @jhaniv ☝️

File: src/provider/handler/mappers/mapPostMediaRequest.js
Comment (francois-codes) [reply src/provider/handler/mappers/mapPostMediaRequest.js]: If you return `{ id: null }` here, it means that where you run `Promise.all(mapPostMediaRequest)`, one of the mediaItems will be `{ id: null }`, which may cause an issue cause you're not ruling out this case in the response processing. If you `return null` then I think it would remove the faulty item out of the array that contains all the promises

**Comment by jhaniv** [comment]
@f-roland ok - all fixes should be done now

File: src/provider/handler/mappers/postMapper.js
Comment (francois-codes) [inline src/provider/handler/mappers/postMapper.js]: 2 thinks here : 
* you can filter the mediaItem array before map / reduce
* you're transforming an array to an array here, so map is more appropriate

the way you've written it, the final array will have only one item, cause you are disregarding the result when returning. reduce doesn't automatically append the returned value, it acts as an accumulator

**Review by francois-codes — CHANGES_REQUESTED** [review]
there's still a problem with the construction of the media group

File: src/provider/handler/mappers/postMapper.js
Comment (jhaniv) [reply src/provider/handler/mappers/postMapper.js]: yes, the point here is to always return an array with 1 item (or empty) - the reduce is just used to loop through the array and find a match for the post id (if exists). only one match is possible.

Comment (francois-codes) [reply src/provider/handler/mappers/postMapper.js]: Ok, I see.
But I still think that filter / map is more expressive here.
Your logic is : 
- find the item (you can even use array.find to make sure you get only one match)
- transform (with map)

so using both methods is better - especially if you pick find cause then you don't need to iterate over the array once the match is found

Comment (jhaniv) [reply src/provider/handler/mappers/postMapper.js]: ok, used find (no need to map here)

Comment (francois-codes) [reply src/provider/handler/mappers/postMapper.js]: find expects a predicate, not a transformation : https://www.w3schools.com/jsref/jsref_find.asp

it will return you one item. so indeed, you don't need to map after that, but that will return you the original item. you need to apply the transformation after that

**Comment by jhaniv** [comment]
fixed

**Review by francois-codes — APPROVED** [review]

**Pull request merged**
